File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.cpp
// Last updated: 21/07/2026, 23:12:21
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
        int cnt=0;
        for(int i=0;i<nums.size();i++){
            int sum=0;
            for(int j=i;j<nums.size();j++){ // j=i because we have to check every possible sum equal to k. it gives 
                sum+=nums[j];               // starting point and ending point in array.
                if(sum==k){
                    cnt++;
                }
            }
        
        }
        return cnt;
    }
};

File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.java
// Last updated: 21/07/2026, 23:12:15
class Solution {
    public int subarraySum(int[] arr, int k) {
        HashMap<Integer, Integer> mpp = new HashMap<>();

        mpp.put(0,1);

        int cnt =0;
        int sum =0;
        for(int i=0;i<arr.length;i++){

            sum+=arr[i];

            if(mpp.containsKey(sum-k)){
                cnt+=mpp.get(sum-k);
            }
                mpp.put(sum,mpp.getOrDefault(sum,0)+1);
            
        }

        return cnt;
        
    }
}